torch/optim/optimizer.py

```diff
@@ -551,6 +727,13 @@ def update_group(group: Dict[str, Any], new_group: Dict[str, Any]) -> Dict[str,
             update_group(g, ng) for g, ng in zip(groups, saved_groups)]
         self.__setstate__({'state': state, 'param_groups': param_groups})
 
+        for post_hook in self._optimizer_load_state_dict_post_hooks.values():
+            out = post_hook(self)
+            assert out is None, (
```

Comment:
Not sure why we would check this one but not the others?
We do this on Module as there is risk of confusion on how to modify the IncompatibleKeys. This is not a risk here right?

Reply:
oh

Reply:
i think i stuck too close to patternmatching